go/vt/vtgate/buffer/shard_buffer.go

```diff
@@ -569,10 +582,32 @@ func (sb *shardBuffer) drain(q []*entry, err error) {
 	sb.timeoutThread.stop()
 
 	start := sb.timeNow()
-	// TODO(mberlin): Parallelize the drain by pumping the data through a channel.
-	for _, e := range q {
-		sb.unblockAndWait(e, err, true /* releaseSlot */, true /* blockingWait */)
+
+	entryCount := len(q)
+	parallelism := min(sb.buf.config.DrainConcurrency, entryCount)
+
+	var wg sync.WaitGroup
+	var rangeCounter atomic.Int64
+
+	wg.Add(parallelism)
+	for i := 0; i < parallelism; i++ {
+		go func() {
+			defer wg.Done()
+			for {
+				idx, ok := parallelRangeIndex(&rangeCounter, entryCount-1)
+				if !ok {
+					break
```

Comment:
Any reason not to return here?

Reply:
break is exactly what I want here. I want the loop iteration to end. in this instance they are functionally equivalent, but "return" requires modifying the function to place any code after the loop.

Reply:
There is no code after the for loop though, which is why I asked.

Reply:
Another way to put it, IF we want to exit the goroutine and return when parallelRangeIndex returns false, then we should explicitly do so. That can always be changed if that's not what we want.

This is also an example of why comments are nice. Why would or wouldn't we want to return there? What might we want to do outside of the for loop? My reading of it was that we want to loop forever or until parallelIRangeIndex tells us to stop. So it caught my eye that we weren't explicitly stopping when it did.